src/main/kotlin/com/msd/robot/application/RobotApplicationService.kt

```diff
@@ -24,15 +27,52 @@ class RobotApplicationService(
         val robotId = moveCommand.robotId
         val playerId = moveCommand.playerUUID
 
-        val robot =
-            robotRepo.findByIdOrNull(robotId) ?: throw RobotNotFoundException("Can't find robot with id $robotId")
+        val robot = getRobot(robotId)
 
-        if (robot.player != playerId) throw InvalidPlayerException("Specified player doesn't match player specified in robot")
+        checkPlayers(robot.player, playerId)
         val planetDto =
             gameMapService.retrieveTargetPlanetIfRobotCanReach(robot.planet.planetId, moveCommand.targetPlanetUUID)
         val cost = planetDto.movementCost
         val planet = planetDto.toPlanet()
         robot.move(planet, cost)
         robotRepo.save(robot)
     }
+
+    /**
+     * Makes the [Robot] specified in the [BlockCommand] block its current [Planet].
+     *
+     * @param blockCommand            The `BlockCommand` which specifies which robot should block
+     * @throws RobotNotFoundException  if no robot with the ID specified in the `BlockCommand` can be found
+     * @throws InvalidPlayerException  if the PlayerIDs specified in the `BlockCommand` and `Robot` don't match
+     */
+    fun block(blockCommand: BlockCommand) {
+        val robot = getRobot(blockCommand.robotId)
+        checkPlayers(robot.player, blockCommand.playerUUID)
+        robot.block()
+        robotRepo.save(robot)
+    }
+
+    /**
+     * Checks if the two specified PlayerIDs match.
+     *
+     * @param robotPlayerId   the `playerId` of the [Robot]
+     * @param commandPlayerId the `playerUUID` of the [Command]
+     * @return `true` if the IDs match
+     * @throws InvalidPlayerException if both `UUIDs` don't match
+     */
+    private fun checkPlayers(robotPlayerId: UUID, commandPlayerId: UUID) {
```

Comment:
put  this in the domain service too, but make the first parameter a robot instead of a robotPlayerId and rename it to checkIsOwningPlayer or something along those lines